pySDC/implementations/problem_classes/RayleighBenard.py

```diff
@@ -0,0 +1,521 @@
+import numpy as np
+from mpi4py import MPI
+
+from pySDC.implementations.problem_classes.generic_spectral import GenericSpectralLinear
+from pySDC.implementations.datatype_classes.mesh import mesh, imex_mesh
+from pySDC.core.convergence_controller import ConvergenceController
+from pySDC.core.hooks import Hooks
+from pySDC.implementations.convergence_controller_classes.check_convergence import CheckConvergence
+
+
+class RayleighBenard(GenericSpectralLinear):
+    """
+    Rayleigh-Benard Convection is a variation of incompressible Navier-Stokes. See, for instance https://doi.org/10.1007/s00791-020-00332-3.
```

Comment:
At this point, please write the exact equations you are solving here ... giving an external reference to describe what you are doing here is vicious